addons/l10n_ch/models/res_bank.py

```diff
@@ -35,6 +51,39 @@ class ResPartnerBank(models.Model):
     l10n_ch_isr_subscription_eur = fields.Char(string='EUR ISR Subscription Number', help='The subscription number provided by the bank or Postfinance to identify the bank, used to generate ISR in EUR. eg. 03-162-5')
     l10n_ch_show_subscription = fields.Boolean(compute='_compute_l10n_ch_show_subscription', default=lambda self: self.env.company.country_id.code == 'CH')
 
+    def is_isr_issuer(self):
+        return (_is_l10n_ch_isr_issuer(self.l10n_ch_postal, 'chf')
+                or _is_l10n_ch_isr_issuer(self.l10n_ch_postal, 'eur'))
+
+    @api.constrains("l10n_ch_postal")
+    def _check_postal_num(self):
+        """Validate postal number format"""
+        for rec in self:
+            if rec.l10n_ch_postal and not _is_l10n_ch_postal(self.l10n_ch_postal):
```

Comment:
@yvaucher Do not we need to add the condition then that this only applies for the non-company bank accounts?

Reply:
This is in line with #51637 where `l10n_ch_postal` has no use for your company bank account (field made invisible).

As explained multiple times it's a design flaw to have something else in that field than a postal format and makes it overly complicated.

Reply:
I agree it is better to make things clear, but this is overkill.  We are not going to change it that way in v13.